src/test/kotlin/org/rust/ide/annotator/RsUnsafeExpressionHighlightingAnnotatorTest.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import org.rust.ide.colors.RsColor
+
+class RsUnsafeExpressionHighlightingAnnotatorTest : RsAnnotatorTestBase(RsUnsafeExpressionAnnotator::class.java) {
+    override fun setUp() {
+        super.setUp()
+        registerSeverities(listOf(RsColor.UNSAFE_CODE.testSeverity))
+    }
+
+    fun `test extern static in unsafe block`() = checkHighlighting("""
+        extern {
+            static FOO: i32;
+        }
+
+        fn main() {
+            unsafe { let a = <UNSAFE_CODE descr="Use of unsafe extern static">FOO</UNSAFE_CODE>; }
+        }
+    """)
+
+    fun `test mutable static in unsafe block`() = checkHighlighting("""
+        static mut FOO : u8 = 0;
+
+        fn main() {
+            unsafe { <UNSAFE_CODE descr="Use of unsafe mutable static">FOO</UNSAFE_CODE> += 1; }
+        }
+    """)
+
+    fun `test unsafe call`() = checkHighlighting("""
+        unsafe fn foo() {}
+        unsafe fn bar() { <UNSAFE_CODE descr="Call to unsafe function">foo</UNSAFE_CODE>(); }
+    """)
+
+    fun `test unsafe call with complex path`() = checkHighlighting("""
+        mod foo {
+            unsafe fn foo<T>() {}
+        }
+        unsafe fn bar() { <UNSAFE_CODE descr="Call to unsafe function">foo::foo::<u8></UNSAFE_CODE>(); }
```

Comment:
Maybe it's worth to narrow highlighting range from `foo::foo::<u8>` to `foo::<u8>`?